File: example/src/main/java/ru/virgil/example/security/SecurityUserDetails.java
package ru.virgil.example.security;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import ru.virgil.example.system.IdentifiedEntity;
import ru.virgil.security.entity.SecurityUser;

import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import java.util.Collection;
import java.util.HashSet;
import java.util.Set;
import java.util.stream.Collectors;

@Entity
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
public class SecurityUserDetails extends IdentifiedEntity implements SecurityUser {

    @ElementCollection
    private Set<UserAuthority> authorities = new HashSet<>();
    private boolean banned;
    private String firebaseUserId;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities.stream()
                .map(userAuthority -> new SimpleGrantedAuthority(userAuthority.name()))
                .collect(Collectors.toSet());
    }

}

File: example/src/main/java/ru/virgil/example/security/SecurityUserDetails.kt
package ru.virgil.example.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import ru.virgil.example.system.entity.BaseEntity
import ru.virgil.example.system.entity.IdentifiedEntity
import ru.virgil.security.entity.SecurityUser
import java.io.Serializable
import java.util.*
import java.util.stream.Collectors
import javax.persistence.*

/**
 * Не наследуется от [IdentifiedEntity] из-за бага Detached Entity.
 */
@Entity
class SecurityUserDetails(
    @Id
    @GeneratedValue
    var uuid: UUID = UUID.randomUUID(),
    @ElementCollection(fetch = FetchType.EAGER)
    var userAuthorities: MutableSet<UserAuthority>,
    override var banned: Boolean = false,
    override var firebaseUserId: String,
) : BaseEntity(), SecurityUser, Serializable {

    override val authorities: Collection<GrantedAuthority>
        get() = userAuthorities.stream()
            .map { userAuthority: UserAuthority -> SimpleGrantedAuthority(userAuthority.name) }
            .collect(Collectors.toSet())

}
